core/dbt/graph/selector_methods.py

```diff
@@ -53,13 +53,18 @@ class MethodName(StrEnum):
     Version = "version"
 
 
-def is_selected_node(
-    fqn: List[str],
-    node_selector: str,
-) -> bool:
+def is_selected_node(fqn: List[str], node_selector: str, is_versioned: bool) -> bool:
     # If qualified_name exactly matches model name (fqn's leaf), return True
-    if fqn[-1] == node_selector:
-        return True
+    if is_versioned:
+        flat_node_selector = node_selector.split(".")
+        if fqn[-2] == node_selector:
+            return True
+        # If this is a versioned model, then the last two segments should be allowed to exactly match
+        elif fqn[-2:] == flat_node_selector[-2:]:
+            return True
```

Comment:
Nice! Ok, this roughly approximates the logic in the docs site now (https://github.com/dbt-labs/dbt-docs/pull/411).

The only differences:
- In the docs site we support selecting as `model_name_v1`, in addition to `model_name.v1`, simply because that's what we visually display in the docs site. This doesn't feel like a massive inconsistency to me.
- In the docs site we support selecting with _just_ `v1`. We could support that here too, by allowing `fqn[-1] == node_selector` for versioned models as well (rather than in an `else` clause below). But this works if I select with the wildcard `*.v1`, and tbh I almost like that better (more explicit).

In both cases, I think I prefer the behavior here, and it should define the standard. The small differences in `dbt-docs` are just there for expediency.